extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -18,34 +18,203 @@ import {CSS} from '../../../build/amp-consent-0.1.css';
+    this.consentUIRequired_ = {};
   }
 
   buildCallback() {
-    if (!isExperimentOn(this.ampdoc.win, AMP_CONSENT_EXPERIMENT)) {
+    if (!isExperimentOn(this.win, AMP_CONSENT_EXPERIMENT)) {
       return;
     }
+
+    // TODO: Decide what to do with incorrect configuration.
+    this.assertAndParseConfig_();
+
+    getServicePromiseForDoc(this.getAmpDoc(), CONSENT_POLICY_MANGER)
+        .then(manager => {
+          this.consentPolicyManager_ = manager;
+          this.generateDefaultPolicy_();
+          const policyKeys = Object.keys(this.policyConfig_);
+          for (let i = 0; i < policyKeys.length; i++) {
+            this.consentPolicyManager_.registerConsentPolicyInstance(
+                policyKeys[i], this.policyConfig_[policyKeys[i]]);
+          }
+        });
+
+    getServicePromiseForDoc(this.getAmpDoc(), CONSENT_STATE_MANAGER)
+        .then(manager => {
+          this.consentStateManager_ = manager;
+          this.init_();
+        });
+
   }
 
   /** @override */
   isLayoutSupported(layout) {
     return layout == Layout.NODISPLAY;
   }
+
+  /**
+   * Init the amp-consent by registering and initiate consent instance.
+   */
+  init_() {
+    this.cidPromise_ = Services.cidForDoc(this.getAmpDoc()).then(cid => {
+      // Note: do not wait for consent of the cid used for amp-consent
+      return cid.get({
+        scope: TAG,
+        createCookieIfNotPresent: true,
+      }, Promise.resolve());
+    });
+
+    const instanceKeys = Object.keys(this.consentConfig_);
+    for (let i = 0; i < instanceKeys.length; i++) {
+      const instanceId = instanceKeys[i];
+      this.consentStateManager_.registerConsentInstance(instanceId);
+      this.getConsentRemote_(instanceId).then(response => {
+        this.parseConsentResponse_(instanceId, response);
+        this.handleUI_(instanceId);
+      }).catch(unusedError => {
+        // TODO: Handle errors
+      });
+    }
+  }
+
+  /**
+   * Generate default consent policy if not defined
+   */
+  generateDefaultPolicy_() {
+    if (this.policyConfig_ && this.policyConfig_['default']) {
+      return;
+    }
+    // Generate default policy
+    const instanceKeys = Object.keys(this.consentConfig_);
+    const defaultWaitForItems = {};
+    for (let i = 0; i < instanceKeys.length; i++) {
+      // TODO: Need to support an array.
+      defaultWaitForItems[instanceKeys[i]] = undefined;
+    }
+    const defaultPolicy = {
+      'itemsToWait': defaultWaitForItems,
+    };
+    this.policyConfig_['default'] = defaultPolicy;
+  }
+
+  /**
+   * Get localStored consent info, and send request to get consent from endpoint
+   * @param {string} instanceId
+   * @return {!Promise<!JsonObject>}
+   */
+  getConsentRemote_(instanceId) {
+    return this.consentStateManager_.getConsentInstanceState(instanceId)
+        .then(localState => {
+          return this.cidPromise_.then(ampUserId => {
+            const request = /** @type {!JsonObject} */ ({
+              'ampUserId': ampUserId,
```

Comment:
clientId

Reply:
Code removed. ClientId is not provided in v1